### ISSUE TITLE
multicast opt dropped in v2 plugin

### ISSUE THREAD

**Issue by lukemarsden** [weaveworks/weave#2816]
```
docker plugin install bboreham/weave2:latest
[...]
```
```
docker network create --driver=bboreham/weave2:latest --opt works.weave.multicast=true wnet69
m1eyxeah2jky7y2cq92yysamh
```
meanwhile, stracing the plugin:
```
luke@beast:~$ sudo strace -s 4096 -f -p 21170
[...]
[pid 21183] read(6, "POST /NetworkDriver.AllocateNetwork HTTP/1.1\r\nHost: \r\nUser-Agent: Go-http-client/1.1\r\nContent-Length: 179\r\nAccept: application/vnd.docker.plugins.v1.2+json\r\n\r\n{\"NetworkID\":\"m1eyxeah2jky7y2cq92yysamh\",\"Options\":{\"works.weave.multicast\":\"true\"},\"IPv4Data\":[{\"AddressSpace\":\"\",\"Gateway\":\"10.0.0.1/24\",\"Pool\":\"10.0.0.0/24\"}],\"IPv6Data\":null}\n", 4096) = 338
[pid 21173] select(0, NULL, NULL, NULL, {0, 20} <unfinished ...>
[pid 21183] write(2, "INFO: 2017/02/24 10:56:49.476811 [net] NetworkAllocate m1eyxeah2jky7y2cq92yysamh\n", 81 <unfinished ...>
[pid 21173] <... select resumed> )      = 0 (Timeout)
[pid 21183] <... write resumed> )       = 81
[pid 21173] select(0, NULL, NULL, NULL, {0, 20} <unfinished ...>
[pid 21183] write(6, "HTTP/1.1 200 OK\r\nDate: Fri, 24 Feb 2017 10:56:49 GMT\r\nContent-Length: 5\r\nContent-Type: text/plain; charset=utf-8\r\n\r\nnull\n", 121) = 121
[...]
```
note `\"Options\":{\"works.weave.multicast\":\"true\"}`
yet:
```
root@beast:/var/run/docker/plugins# docker network inspect wnet69
[
    {
        "Name": "wnet69",
        "Id": "m1eyxeah2jky7y2cq92yysamh",
        "Created": "0001-01-01T00:00:00Z",
        "Scope": "swarm",
        "Driver": "bboreham/weave2:latest",
        "EnableIPv6": false,
        "IPAM": {
            "Driver": "default",
            "Options": null,
            "Config": []
        },
        "Internal": false,
        "Attachable": false,
        "Containers": null,
        "Options": null, # <--
        "Labels": null
    }
]
```
and multicast doesn't work (tested with [mlist](https://github.com/lukemarsden/mballs/tree/mlist)) on containers connected to this network

**Comment by brb** [comment]
The bug was that `NetworkAllocate` (run on Swarm master nodes) returned empty options which meant that:

- Empty options were passed to `CreateNetwork` (run on Swarm worker nodes).
- `NetworkInfo` returned no multicast option when joining an endpoint.

**Comment by marccarre** [comment]
Fixed by #2727.

**Comment by elffirein** [comment]
hi,

I have tried to create weave multicast network in docker swarm using weave V2 plugin. I have executed following command:

> sudo docker network create --driver=weaveworks/net-plugin:latest_release --attachable --opt works.weave.multicast=true mynetwork

Then I have tried to attach containers to this network:
sudo docker run -it --network=mynetwork cazcade/weave-multicast-test (sender)
sudo docker run -it --network=mynetwork cazcade/weave-multicast-test (receiver)

And then started typing message in sender docker instance, but nothing is received. Is the multicast setup right? Do I need to do anymore configuartion changes?

Regards,
Kaustuv
